sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/ExtractBenchmark.scala

```diff
@@ -56,9 +56,11 @@ object ExtractBenchmark extends SqlBasedBenchmark {
     }
   }
 
-  private def castExpr(from: String): String = from match {
+  private def castExpr(from: String, toStr: Boolean = false): String = from match {
     case "timestamp" => "cast(id as timestamp)"
     case "date" => "cast(cast(id as timestamp) as date)"
+    case "interval" if toStr => "cast((cast(cast(id as timestamp) as date) - date'0001-01-01') + " +
```

Comment:
why we need to cast to string?

Reply:
Hmm, in this benchmark, there are 2 kinds of tests, one is `extract/date_part` and the other is `string to interval`, the latter fails type checking when saving out.
Or shall we just remove the `string to interval` case here https://github.com/apache/spark/pull/27867/files#diff-b89131adf146fef7cd6e3db381fd0807L111

Reply:
Why it's not a problem for date/timestamp?

Reply:
We forbid interval as output schema only

Reply:
Maybe we should not use the noop sink here, but `df.queryExecution.toRDD.foreach(_ => ())`

Reply:
good advice, thanks!